core/src/main/java/nu/yona/server/rest/ErrorLoggingFilter.java

```diff
@@ -72,6 +86,52 @@ private static String getCorrelationId(HttpServletRequest request)
 			return traceIdHeader;
 		}
 
+		private static void putAppVersionContext(String header)
+		{
+			String[] parts = header.split("/");
+			String operatingSystem = parts[0];
+			int versionCode = Integer.parseInt(parts[1]);
+			String versionName = parts[2];
+			MDC.put(APP_OS_MDC_KEY, operatingSystem);
+			MDC.put(APP_VERSION_CODE_MDC_KEY, versionCode);
+			MDC.put(APP_VERSION_NAME_MDC_KEY, versionName);
+		}
+
+		private static void assertValidHeaders(HttpServletRequest request)
+		{
+			Optional.ofNullable(request.getHeader(Constants.APP_VERSION_HEADER))
+					.ifPresent(LoggingContext::validateAppVersionHeader);
+		}
+
+		private static void validateAppVersionHeader(String header)
+		{
+			String[] parts = header.split("/");
+			Require.that(parts.length == 3, () -> InvalidDataException.invalidAppVersionHeader(header));
+			assertValidOperatingSystem(parts[0]);
```

Comment:
This puts quite strict requirements on the app version header. Can we test this in time that it does not block the app?

Reply:
Yes, on the beta server